src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/SetOfValueComparer.cs

```diff
@@ -17,6 +17,18 @@ internal static int Compare(ReadOnlySpan<byte> x, ReadOnlySpan<byte> y)
             int min = Math.Min(x.Length, y.Length);
             int diff;
 
+#if NET7_0_OR_GREATER
+            int diffIndex = x.Slice(0, min).CommonPrefixLength(y.Slice(0, min));
```

Comment:
You shouldn't need to do this slicing.  What happens if you don't?

Reply:
> What happens if you don't?

Uh, everything works as expected. So... removed the `Slice`s.

Reply:
But now I see why this works, thanks for the suggestion.

https://github.com/dotnet/runtime/blob/f95f0c06823bb407717434ee31a68ab06b889dbb/src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs#L2672